src/app/sevices/user.service.ts

```diff
@@ -24,12 +24,29 @@ export class UserService {
         return this.http.get<User[]>(backend_users);
     }
 
-    // GET /users
+    // GET /users/current
     // Returns the current user, based on the JWT token in cookies
     getCurrentUser(): Observable<User> {
         return this.http.get<User>(backend_users + "current");
     }
 
+    // GET /users/<id>
+    // Looks for a user with the provided id
+    getUser(id: string): Observable<User> {
+        return this.http.get<User>(backend_users + id);
+    }
+
+    // PUT /users/<id>
+    // Updates the user with the user data from body,
+    // Returns the updated user
+    updateUser(user: User, userId: string): Observable<User> {
```

Comment:
subscribe within a subscribe within a subscribe, also you are not unsubscribing from this method in the user details component